lib/add_to_calendar.rb

```diff
@@ -27,6 +27,30 @@ def initialize(start_datetime:, end_datetime: nil, title:, timezone:, location:
   
       validate_attributes
     end
+    
+    def hey_url
+      calendar_url = "https://app.hey.com/calendar/ical_events/new?ical_source=BEGIN%3AVCALENDAR%0AVERSION%3A2.0%0APRODID%3A-%2F%2FBasecamp%2F%2FBC3%2F%2FEN%0ACALSCALE%3AGREGORIAN%0A%0ABEGIN%3AVEVENT"
+    
+      params = {}
+      params[:SUMMARY] = url_encode_ical(title)
+      params[:DESCRIPTION] = url_encode_ical(description) if description
+      params[:DTSTAMP] = Time.now.strftime("%Y%m%dT%H%M%SZ")
+       params[:DTSTART] = utc_datetime(start_datetime)
+        if end_datetime
+          params[:DTEND] = utc_datetime(end_datetime)
+        else
+          params[:DTEND] = utc_datetime(start_datetime + 60*60) # 1 hour later
+        end
+      params[:UID] = "-#{url_encode(url)}" if url
+      params[:URL] = url_encode(url) if url
+    
+      params.each do |key, value|
+        calendar_url << "%0A#{key}%3A#{value}"
+      end
+    
+      calendar_url << "%0AEND%3AVEVENT%0AEND%3AVCALENDAR"
+      return calendar_url
```

Comment:
To build the URL with the query, you could build the ical source as regular text and then do:

```ruby
query = URI.encode_www_form(ical_source: ical_source)
"https://app.hey.com/calendar/ical_events/new?#{query}"
```

So that you don't have to care about escaping characters and such.

Reply:
Thanks @jorgemanrubia  I'm having issues using the URI.encode_www_form. Tried to get some hints but I'm unfortunately can't get it to print the params out separated with %0A. Any hint you can give me around how you do in BC?